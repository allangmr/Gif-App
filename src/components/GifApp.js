import { useState } from 'react'
import AddCategory from './AddCategory';
import GifGrid from './GifGrid';

const GifApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <div>
            <h2>GifApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr />
            <ol> 
                {
                    categories.map(category => (
                        <GifGrid key={category} category={category} />
                    ))
                }
            </ol>
        </div>
    )
}

export default GifApp
