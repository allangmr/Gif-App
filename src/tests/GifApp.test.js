import { shallow } from "enzyme";
import GifApp from "../components/GifApp";

describe('Test of GifApp.js', () => {
    test('should be a component snapshot', () => {
        const wrapper = shallow(<GifApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('should be show a category list', () =>{
        const categories = ['Developer', 'Happy'];
        const wrapper = shallow(<GifApp  defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
})
