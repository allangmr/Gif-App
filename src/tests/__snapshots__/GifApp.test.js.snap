// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of GifApp.js should be a component snapshot 1`] = `
<div>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Test of GifApp.js should be show a category list 1`] = `
<div>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Developer"
      key="Developer"
    />
    <GifGrid
      category="Happy"
      key="Happy"
    />
  </ol>
</div>
`;
